From 06a9e39b933d699c903d230ac82429b83bf02c0e Mon Sep 17 00:00:00 2001
From: Alex Richardson <alexrichardson@google.com>
Date: Mon, 5 Oct 2026 12:41:26 -0700
Subject: [PATCH] Fix RV32 build of the invalid address self test

The 64-bit literal cannot be zero-extended to the RV32 capability length.
---
 model/cheri/test_main_ext.sail | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/model/cheri/test_main_ext.sail b/model/cheri/test_main_ext.sail
index 6244fbbeb4d4568cce053f60cc7a3474a7c7fb21..0c32592b38b76a9f2a3ea1ca78b1dbc92ab47f08 100644
--- a/model/cheri/test_main_ext.sail
+++ b/model/cheri/test_main_ext.sail
@@ -501,7 +501,7 @@ function test_cheri_invalid_addr() -> unit = {
     // If system has no invalid addresses, escape
     if (validAddr(invalidAddr)) then return ();
     // Since invalid addr checks are waived for infinite caps, make a slightly smaller cap
-    let (_, finite_cap) = setCapBounds(infinite_cap, zeros(), zero_extend(0xff00_0000_0000_0000));
+    let (_, finite_cap) = setCapBounds(infinite_cap, zeros(), zero_extend(0xff @ zeros(xlen - 8)));
     let infiniteR : regidx = Regidx(0b00001);
     let finiteR : regidx = Regidx(0b00010);
     let zeroR : regidx = Regidx(0b00000);
-- 
2.56.0.rc1.315.gc6ed9934b7-goog

